validator/src/main/java/com/amazon/aoc/validators/AlarmPullingValidator.java

```diff
@@ -64,11 +76,14 @@ public void validate() throws Exception {
                   metricAlarm.getAlarmName(),
                   metricAlarm.getStateValue(),
                   metricAlarm.getMetrics());
+              cloudWatchService.putMetricData(SOAKING_NAMESPACE, METRIC_NAME, 0.0, dimension);
               System.exit(1);
             }
           }
 
-          log.info("alarms look good, continue to bake");
+          log.info("all alarms look good, continue to bake");
+          cloudWatchService.putMetricData(SOAKING_NAMESPACE, METRIC_NAME, 1.0, dimension);
```

Comment:
i think we just emit one metric after all the retries, but looks like we are going to emit metrics for every retry?

Reply:
yes, it will emit metrics for every pulling alarm check. I feel not too much difference on emitting the metrics for each retry and all retries. For every retry gets better granularity on the metrics.